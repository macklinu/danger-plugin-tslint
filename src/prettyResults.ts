import { oneLine } from 'common-tags'
import { IRuleFailureJson, IRuleFailurePositionJson } from 'tslint'

const lineAndCharacter = (position: IRuleFailurePositionJson) => {
  const line = position.line + 1
  const character = position.character + 1
  return `${line}, ${character}`
}

export function prettyResults(results: IRuleFailureJson[]): string {
  return results
    .map((result) => oneLine`
      ${result.ruleSeverity}:
      ${result.name}[${lineAndCharacter(result.startPosition)}]:
      ${result.failure}
    `)
    .map((item: string) => `- ${item}`)
    .join('\n')
}
