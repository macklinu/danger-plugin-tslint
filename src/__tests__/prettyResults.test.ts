import { prettyResults } from '../prettyResults'

test('it formats failures', () => {
  const failureResults = require('./__fixtures__/failure-results.json')
  expect(prettyResults(failureResults)).toMatchSnapshot()
})
