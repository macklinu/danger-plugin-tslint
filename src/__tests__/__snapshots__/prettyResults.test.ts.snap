// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it formats failures 1`] = `
"- ERROR: src/index.ts[48, 57]: Unnecessary semicolon
- ERROR: src/index.ts[49, 38]: Unnecessary semicolon
- ERROR: src/index.ts[53, 18]: Unnecessary semicolon"
`;
